src/main/java/com/ashkan/moviesapi/entities/MovieCatalog.java

```diff
@@ -18,7 +22,7 @@
     private Price priceByPriceId;
 
     @Id
-    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```

Comment:
According to the requirements the PK of this Entity should be the object Movie.
You need to use that Entity as a PK and FK at the same time, please take a look how to do that.